controllers/repository_controller.go

```diff
@@ -161,7 +161,12 @@ func (r *RepositoryReconciler) UpdateRepository(ctx context.Context, logger logr
 		}
 		return false, err
 	}
-
+	if instanceDeepCopy.Annotations == nil {
+		instanceDeepCopy.Annotations = make(map[string]string)
+	}
+	if v, ok := instanceDeepCopy.Annotations[corev1alpha1.RepositorySourceAnnotation]; !ok || (v != string(corev1alpha1.Opensource) && v != string(corev1alpha1.Official) && v != string(corev1alpha1.Internal)) {
+		instanceDeepCopy.Annotations[corev1alpha1.RepositorySourceAnnotation] = string(corev1alpha1.Unknown)
+	}
```

Comment:
After setting up annotations, update the instance.

Reply:
https://github.com/kubebb/core/blob/d1e38921242e4c170ea039d2a4eeb4d645179f40/controllers/repository_controller.go#L183

When the annotations are updated, there will be a new update event, but the judgment conditions here are not updated, so it will not enter Reconcile processing.

So here we need to add a judgment on annotations.

Reply:
Another question, why isn't the type added to labels? May need to search depending on the type of repository

Reply:
> Another question, why isn't the type added to labels? May need to search depending on the type of repository.

sounds good.

Reply:
there is no special reasons. pls move this to labels so we can group same kind of repositories together.